src/main/java/coresearch/cvurl/io/sse/EventParser.java

```diff
@@ -0,0 +1,87 @@
+package coresearch.cvurl.io.sse;
+
+import coresearch.cvurl.io.mapper.GenericMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+
+class EventParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EventParser.class);
+    private static final String ID = "id";
+    private static final String EVENT = "event";
+    private static final String RETRY = "retry";
+    private static final String DATA = "data";
+    private static final String COLON = ":";
+    private static final String LINE_FEED = "\n";
+    private static final String NULL = "\0";
+    private static final String DIGITS_ONLY = "^\\d*$";
+
+    private final SseEventListener listener;
+    private final GenericMapper genericMapper;
+
+    EventParser(SseEventListener listener, GenericMapper genericMapper) {
+        this.listener = listener;
+        this.genericMapper = genericMapper;
+    }
+
+    public void parse(InputStream eventStream) {
+        if (eventStream == null)
+            return;
+
+        String line;
+        InboundServerEvent.InboundServerEventBuilder builder = InboundServerEvent.newBuilder(genericMapper);
+        try (final BufferedReader reader = new BufferedReader(new InputStreamReader(eventStream, StandardCharsets.UTF_8))) {
+            while ((line = reader.readLine()) != null) {
+                if (line.isEmpty()) {
+                    listener.onEvent(builder.build());
+                    builder = InboundServerEvent.newBuilder(genericMapper);
+                    continue;
+                }
+
+                if (line.startsWith(COLON))
+                    continue;
+
+                final String[] parts = line.split(COLON, 2);
+                final String field = parts[0];
+                final String value = removeOneLeadingSpace(parts.length > 1 ? parts[1] : "");
+                if (field.equals(EVENT))
+                    builder.name(value);
+
+                if (field.equals(DATA)) {
+                    if (builder.getData() != null)
+                        builder.data(builder.getData() + LINE_FEED + value);
+                    else
+                        builder.data(value);
+                }
+
+                if (field.equals(ID) && !value.contains(NULL))
```

Comment:
NPE?

Reply:
Here I check for U+0000 (NULL) character in the value of Id field:

> If the field name is "id"
If the field value does not contain U+0000 NULL, then set the last event ID buffer to the field value. Otherwise, ignore the field.

Reply:
Will rename constant to `NULL_CHARACTER` to avoid confusion